pages/02.general/02.how-to-create-your-own-ad-filters/docs.en.md

```diff
@@ -971,9 +1221,90 @@ example.com#@%#window.__gaq = undefined;
 
 If you maintain a third-party filter that is known to AdGuard, you might be interested in the information presented in this section. Please note, that hints will be applied to registered filters only. The filter is considered to be registered and known by AdGuard, if it is present in the [known filters index](https://filters.adtidy.org/extension/chromium/filters.json).  If you want your filter to be registered, please file an issue to [AdguardFilters repo](https://github.com/AdguardTeam/AdguardFilters).
 
+<a id="pre_processor"></a>
+### Pre-processor directives
+
+We provide multiple pre-processor directives that can be used by filters maintainers to improve compatibility with different ad blockers.
+
+#### Syntax
+
+```
+!#if condition
+Anything goes here
+!#include URL_or_a_relative_path
+!#endif
+```
+
+* `!#if`, `!#endif` -- filters maintainers can use these conditions to supply different rules depending on the ad blocker type.
+* `condition` -- just like in some popular programming languages, pre-processor conditions are based on constants declared by ad blockers. Ad blocker authors define on their own what exact constants do they declare.
+* `!#include` -- this directive allows to include contents of a specified file into the filter.
+
+#### Conditions
+
+When an adblocker encounters an `!#if` directive, followed eventually by an `!#endif` directive, it will compile the code between the directives only if the specified condition is true. Condition supports all the basic logical operators.
+
+**Example**:
+```
+!#if (adguard && !adguard_ext_safari)
+||example.org^$third-party
+!#endif
+```
+
+#### Including a file
+
+The `!#include` directive supports only files from the same origin to make sure that the filter maintainer is in control of the specified file. The included file can also contain pre-processor directives (even other `!#include` directives).
+
+Ad blockers should consider the case of recursive `!#include` and implement a protection mechanism.
+
+**Examples**
+
+Filter URL: `https://example.org/path/filter.txt`
+```
+!
+! Valid (same origin):
+!#include https://example.org/path/includedfile.txt
+!
+! Valid (relative path):
+!#include /includedfile.txt
+!#include ../path2/includedfile.txt
+!
+! Invalid (another origin):
+!#include https://example.com/path/includedfile.txt
+```
+
+#### Remarks
+
+* If included file is not found or unavailable, the whole filter update should fail.
+* A conditional directive beginning with a #if directive must explicitly be terminated with a #endif directive.
```

Comment:
...with an `!#if` directive..., ...with an `!#endif` directive...

to kkep consistency